src/js/FocusTracker.js

```diff
@@ -0,0 +1,99 @@
+'use strict'
+
+/**
+ * @constructor FocusTracker
+ * A custom focus tracker for a DOM element with complex internal DOM structure
+ * @param  {[Object]} config    A set of configurations for the FocusTracker
+ *                {DOM Object} target    The DOM object to track
+ *                {Function} onFocus    onFocus callback
+ *                {Function} onBlur    onBlur callback
+ *
+ * @return
+ */
+
+export class FocusTracker {
+  constructor (config) {
+    this.onFocus = (typeof config.onFocus === 'function') ? config.onFocus : null
+    this.onBlur = (typeof config.onBlur === 'function') ? config.onBlur : null
+    this.target = config.target || null
+    this._onClick = this._onEvent.bind(this)
+    this._onKeyUp = function (event) {
+      if (event.which === 9 || event.keyCode === 9) {
+        this._onEvent(event)
+      }
+    }.bind(this)
+
+    this.focusFlag = false
+    this.trackerSetFlag = false
+    this.firstEventFlag = true
+
+    /*
+      Adds required (click and keyup) event listeners to the 'document' object
+      to track the focus of the given 'target'
+     */
+    if (!this.trackerSetFlag) {
```

Comment:
I think this `if (!this.trackerSetFlag)` is redundant now :) 

Also, I don't think `FocusTracker` is useful at all without `this.target`, so that can probably be just required instead of optional?

Reply:
Ah yeah my bad :sweat_smile:, I will remove the dependency on the `this.trackerSetFlag` . Also to make `this.target` required I am adding a check for its availability and throwing an error if its not there.